plugin-feature-doc.md

```diff
@@ -0,0 +1,111 @@
+# Plugin API is simple and well-documented
+
+The current plugin API has an overly large surface area, poor documentation,
+and unclear names.  The objective is to improve this, focusing on the UX of
+primary use-case: creating new service authenticators.
+
+Aha Card: https://cyberark.aha.io/features/AAM-92
+
+- [Objective](#objective)
+- [Experience (Summary)](#experience-summary)
+- [Experience (Detailed)](#experience-detailed)
+- [Technical Details](#technical-details)
+- [Testing](#testing)
+- [Documentation](#documentation)
+- [Open Questions](#open-questions)
+- [Stories](#stories)
+- [Future Work](#future-work)
+
+### Objective
+
+- Simplify Plugin API so authors of new service authenticators have a good
+  experience which requires learning only concepts relevant to their goal.
+- Write clear documentation to make the simplified API accessible to new
+  contributors
+
+### Experience (Summary)
+
+The new plugin API will support writing new service authenticators only.  Only
+a minimal interface will be required.
+
+### Experience (Detailed)
+
+Authors will need to implement only `PluginInfo` and a new
+`ServiceAuthenticator` interface.
+
+`APIVersion` will be moved into `PluginInfo`, and the `plugin_type` will be
+explicitly specified:
+
+```go
+var PluginInfo = map[string]string{
+  "api_version": "2.0",
+  "plugin_types": "service_authenticator",
+  "version":     "0.0.7",
+  "id":          "test-plugin",
+  "name":        "Test Plugin",
+  "description": "Test plugin to demonstrate plugin functionality",
+}
+```
+
+Implementing authenticators will look like:
+
+```go
+func ServiceAuthenticators() map[string]func(plugin_v2.AuthenticatorOptions) plugin_v2.Authenticator {
+  return map[string]func(plugin_v2.AuthenticatorOptions) plugin_v2.Authenticator{
+    "sample-authenticator-plugin": samplePlugin.NewAuthenticator,
+  }
+}
+```
+
+With the precise interface of an `Authenticator` TBD.
```

Comment:
Is that a TODO for this document? when will that be defined?

Reply:
Not for this document (at least in my mind).

What I meant there is that when, internally in the refactor, we combine the artists currently called `Listeners` and `Handlers` we'll have a new thing called `Authenticator` and precisely what fields and methods it has will be determined as part of that refactoring.  So I'm considering this an "in the weeds" question that I wasn't going to answer here.